client/main.lua

```diff
@@ -2,9 +2,30 @@ local config = require 'config.client'
 local JOBS = exports.qbx_core:GetJobs()
 local GANGS = exports.qbx_core:GetGangs()
 local isLoggedIn = LocalPlayer.state.isLoggedIn
+local dynamicMenuItems = {}
 
 lib.locale()
 
+-- Adds item to the boss/gang menu.
+---@param menuItem ContextMenuItem Requires args.type to be set to know which menu to place in.
+---@return number menuId ID of the menu item added
+local function addMenuItem(menuItem)
+    local menuId = #dynamicMenuItems + 1
+    if not menuId.args.type then return end
+    dynamicMenuItems[menuId] = lib.table.deepclone(menuItem)
+    return menuId
+end
+exports('AddBossMenuItem', addMenuItem)
+exports('AddGangMenuItem', addMenuItem)
```

Comment:
If Illenium appearance is restarted, won't it add duplicate menu items? Should we check for this and prevent it?

Reply:
illenium-appearance handles this case with https://github.com/iLLeniumStudios/illenium-appearance/blob/2461991d68870721841233cdd5508cf334d3b3ec/client/common.lua#L77-L83.